app/controllers/email_controller.rb

```diff
@@ -46,10 +46,12 @@ def perform_unsubscribe
 
   def unsubscribed
     @email = Discourse.cache.read(params[:key])
-    @topic_id = params[:topic_id]
-    user = User.find_by_email(@email)
-    raise Discourse::NotFound unless user
-    topic = Topic.find_by(id: params[:topic_id].to_i) if @topic_id
-    @topic = topic if topic && Guardian.new(nil).can_see?(topic)
+
+    raise Discourse::NotFound unless User.find_by_email(@email)
+
+    if @topic_id = params[:topic_id]
+      topic = Topic.find_by(id: @topic_id)
+      @topic = topic if topic && Guardian.new.can_see?(topic)
+    end
```

Comment:
Is there any logic change here? Or just an unrelated refactor? 👀

Reply:
No logic changes, just refactor.

Reply:
I see, ok. Ideally this would be split out into a separate `DEV:` PR. Otherwise `git blame` will make it seem like it's related to the implementation of RFC8058, which would be very confusing for future travellers.